test/integ/test_put_get_with_gcp_account.py

```diff
@@ -322,3 +361,133 @@ def test_auto_compress_off_gcp(tmpdir, conn_cnx, db_parameters):
+                    if mocked_get.counter == 0:
+                        mocked_get.counter += 1
+                        exc = requests.exceptions.HTTPError(response=requests.Response())
+                        exc.response.status_code = error_code
+                        raise exc
+                    else:
+                        return get(*args, **kwargs)
+
+                mocked_get.counter = 0
+
+                with mock.patch('snowflake.connector.cursor.SnowflakeFileTransferAgent',
+                                side_effect=mocked_file_agent):
+                    with mock.patch('requests.get', side_effect=mocked_get):
+                        csr.execute("get @%{} file://{}".format(table_name, tmp_dir))
+                    if error_code == 401:
+                        assert mocked_file_agent.agent.renew_expired_client.call_count == 1
+                rec = csr.fetchone()
+                assert rec[0].startswith('data_'), 'A file downloaded by GET'
+                assert rec[1] == 36, 'Return right file size'
+                assert rec[2] == 'DOWNLOADED', 'Return DOWNLOADED status'
+                assert rec[3] == '', 'Return no error message'
+            finally:
+                csr.execute("drop table {}".format(table_name))
+
+    files = glob.glob(os.path.join(tmp_dir, 'data_*'))
+    with gzip.open(files[0], 'rb') as fd:
+        contents = fd.read().decode(UTF8)
+    assert original_contents == contents, 'Output is different from the original file'
+
+
+def test_put_overwrite_with_downscope(tmpdir, conn_cnx, db_parameters):
+    """Tests whether _force_put_overwrite and overwrite=true works as intended."""
+    with conn_cnx() as cnx:
+
+        tmp_dir = str(tmpdir.mkdir('data'))
+        test_data = os.path.join(tmp_dir, 'data.txt')
+        with open(test_data, 'w') as f:
+            f.write("test1,test2")
+            f.write("test3,test4")
+
+        cnx.cursor().execute("RM @~/test_put_overwrite")
+        try:
+            with cnx.cursor() as cur:
+                try:
+                    cur.execute('ALTER SESSION SET GCS_USE_DOWNSCOPED_CREDENTIAL = TRUE')
+                except ProgrammingError:
+                    # skip test if parameter is not available (server not updated)
+                    return
+                with mock.patch.object(cur, '_init_result_and_meta', wraps=cur._init_result_and_meta) as mock_result:
+                    cur.execute("PUT file://{} @~/test_put_overwrite".format(test_data))
+                    assert mock_result.call_args[0][0]['rowset'][0][-2] == 'UPLOADED'
+                with mock.patch.object(cur, '_init_result_and_meta', wraps=cur._init_result_and_meta) as mock_result:
+                    cur.execute("PUT file://{} @~/test_put_overwrite".format(test_data))
+                    assert mock_result.call_args[0][0]['rowset'][0][-2] == 'SKIPPED'
+                with mock.patch.object(cur, '_init_result_and_meta', wraps=cur._init_result_and_meta) as mock_result:
```

Comment:
Why do you mock these? You could just actually get the results

Reply:
Updated, I copied this from s3, did not look at it much